src/javascript/binary/websocket_pages/user/account/authenticate.js

```diff
@@ -23,6 +26,192 @@ const Authenticate = (() => {
+                .find('.remove')
+                .click(() => {
+                    const default_text = event.target.id.split('_')[0]
+                        .replace(/^\w/, w => w.toUpperCase());
+                    $(event.target).val(''); // Remove previously selected file.
+                    $(event.target).parent().find('label').text(default_text)
+                        .append($('<img/>', { class: 'add' }));
+                    // Change submit button state
+                    showSubmit();
+                });
+        };
+
+        /**
+         * Enables the submit button if any file is selected, also adds the event handler for the button.
+         * Disables the button if it no files are selected.
+         */
+        const showSubmit = () => {
+            let file_selected = false;
+            const $ele = $('#authentication-message > div').not('.invisible');
+            $button = $ele.find('#btn_submit');
+            const $files = $ele.find('input[type="file"]');
+
+            // Check if any files are selected or not.
+            $files.each((i, e) => {
+                if (e.files && e.files.length) {
+                    file_selected = true;
+                }
+            });
+
+            if (file_selected) {
+                if ($button.hasClass('button')) return;
+                $button.removeClass('button-disabled')
+                    .addClass('button')
+                    .off('click') // To avoid binding multiple click events
+                    .click(() => submitFiles($files));
+            } else {
+                if ($button.hasClass('button-disabled')) return;
+                $button.off('click')
+                    .removeClass('button')
+                    .addClass('button-disabled');
+            }
+        };
+
+        const disableButton = () => {
+            if ($button.length && !$button.find('.barspinner').length) {
+                const $btn_text = $('<span/>', { text: $button.find('span').text(), class: 'invisible' });
+                showLoadingImage($button.find('span'), 'white');
+                $button.find('span').append($btn_text);
+            }
+        };
+
+        const enableButton = () => {
+            if ($button.length && $button.find('.barspinner').length) {
+                $button.find('>span').html($button.find('>span>span').text());
+            }
+        };
+
+        /**
+         * On submit button click
+         */
+        const submitFiles = ($files) => {
+            // Disable submit button
+            disableButton();
+            const files = [];
+            $files.each((i, e) => {
+                if (e.files && e.files.length) {
+                    const $e = $(e);
+                    const id = $e.attr('id');
+                    const type = `${($e.attr('data-type') || '').replace(/\s/g, '_').toLowerCase()}_${id.split('_')[0]}`;
+                    const $inputs = $e.parent().parent().parent().find('input[type="text"]');
```

Comment:
is there any way to use class or id identifier instead of calling so many `parent()`s? in case structure changes later